runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -237,6 +248,39 @@ private boolean findSharedCore() {
         return true;
     }
 
+    private boolean findDownloadedCore() {
+        String libDir = mWrapperContext.getDir(XWALK_CORE_EXTRACTED_DIR, Context.MODE_PRIVATE).
+                toString();
+        String dexPath = libDir + File.separator + XWALK_CORE_CLASSES_DEX;
+        String dexOutputPath = mWrapperContext.getDir(OPTIMIZED_DEX_DIR, Context.MODE_PRIVATE).
+                getAbsolutePath();
+        ClassLoader localClassLoader = ClassLoader.getSystemClassLoader();
+        mBridgeLoader = new DexClassLoader(dexPath, dexOutputPath, libDir, localClassLoader);
+
+        if (!checkCoreVersion() || !checkCoreArchitecture()) {
+            mBridgeLoader = null;
+            return false;
+        }
+
+        Log.d(TAG, "Running in downloaded mode");
+        mCoreStatus = XWalkLibraryInterface.STATUS_MATCH;
+        return true;
+    }
+
+    private boolean isDownloadMode() {
+        try {
+            PackageManager packageManager = mWrapperContext.getPackageManager();
+            ApplicationInfo appInfo = packageManager.getApplicationInfo(
+                    mWrapperContext.getPackageName(), PackageManager.GET_META_DATA);
+            final String enableStr = appInfo.metaData.getString(META_XWALK_ENABLE_DOWNLOAD_MODE);
+            if (enableStr != null && enableStr.equalsIgnoreCase("enable")) {
```

Comment:
Just `return enableStr.equalsIgnoreCase("enable")` is okay. The `NullPointerException` will be caught.